### ISSUE TITLE
Blender Crashes After Adding FreeCad/lib to Sverchok Settings

### ISSUE THREAD

**Issue by adamhill** [nortikin/sverchok#3594]
## Problem statement
I have Blender 2.9.0_01 on OSX Catalina 15.7, Installed Sverchok, started diagnosing Extras issues. All were loading according to Prefs except for PMCubes and FreeCad. I added FreeCad/libs path to Sverchok preferences, restarted now it crashes immediately when I launch from the command line.

Output from the console.

```/Applications/blender.app/Contents/MacOS/blender                                                                             ─╯
Read prefs: /Users/adamhill/Library/Application Support/Blender/2.90/config/userpref.blend
found bundled python: /Applications/blender.app/Contents/Resources/2.90/python
sv: import settings
PyMCubes package is not available. PyMCubes-based implementation of Marching Cubes will not be available
Writing: /var/folders/tr/1bh5hz6d5559f3249fqcqg880000gn/T/blender.crash.txt
[1]    52103 segmentation fault  /Applications/blender.app/Contents/MacOS/blender```

## Steps to reproduce

1. Install Blender and Sverchok on OSX. Make sure it is running.
2. Install FreeCad, set the path to FreeCad/libs via the Node preferences.
2.5 Set to the correct version of Python 3.7.8 via Pyenv.
2.6 Run Blender from the command line.
3. Crashes immediately.

## Expected result
Blender at least running.

## Actual result
Immediate crash.

## Sverchok version
From sverchok-master.zip from github. Latest FreeCad. Python 3.7.8 via Pyenv. Running blender from the console.

**Comment by vicdoval** [comment]
It seems there is some issue between OSX and FreeCAD...
Are you able to run FreeCAD?
To reset to the previous state you have to delete a file called `freecad_path.pth` from `Blender_Folder\python\lib\site-packages`

**Comment by portnov** [comment]
@adamhill  please try the following:
* Run Blender (under osx, without configured freecad integration)
* Switch to Text Editor window (Shift-F11)
* Paste the following text:
```python
FREECAD = "/your/path/to/freecad/libs"
import sys
sys.path.append(FREECAD)
import FreeCAD
```
* Press Run Script

Will Blender crash?

**Comment by adamhill** [comment]
I think FreeCAD from Homebrew is linked with Python 3.6.  It installs v0.18-16146.

First few stack trace lines are:

```Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libpython3.6m.dylib           	0x0000000128957581 PySys_GetObject + 17
1   FreeCAD.so                    	0x0000000128352054 PyInit_FreeCAD + 964
2   blender                       	0x000000010e483d93 _PyImport_LoadDynamicModuleWithSpec + 579
```
I can run FreeCAD stand alone.

Manually trying to load FreeCAD via script also crashes. Same first lines in the stack trace.
The path I used was '/Applications/FreeCAD.app/Contents/Resources/lib'

I have attached the OSX crash report.

[Sverchok - Blender Crash.txt](https://github.com/nortikin/sverchok/files/5288971/Sverchok.-.Blender.Crash.txt)

**Comment by portnov** [comment]
Uh. Blender is bundled with python3.7. So, no wonder it doesn't want to work with FreeCAD built with python 3.6 or 3.8. Are there builds of FreeCAD for OSX available with Python 3.7?

**Comment by adamhill** [comment]
Ok. Thats what I figured.

So following [this formula](https://forum.freecadweb.org/viewtopic.php?f=42&t=46144&start=10#p397181) I was able to build a Python 3.7 version of FreeCAD. I set the path to this installation and it seems to recognize FreeCAD & not crash - YAY!

I did get some errors in the console, so I don't know exactly what might be broken. 

Is there a FreeCAD blender file I can load or some script I can run to test out all the nodes? Maybe not all, but a good portion.

Since one of the errors is a missing symbol, I am going to see if I can get conda-forge to intall 0.18.4-xxxxx (by default it grabbed v0.19-xxxx.

Here are the errors from the console:

```
Dependencies available: pip, scipy, geomdl, scikit-image, mcubes, circlify, lbt-ladybug, FreeCAD.
sv: import all modules
Traceback (most recent call last):
  File "/Applications/blender.app/Contents/Resources/2.90/scripts/modules/addon_utils.py", line 351, in enable
    mod = __import__(module_name)
  File "/Users/adamhill/Library/Application Support/Blender/2.90/scripts/addons/sverchok-master/__init__.py", line 73, in <module>
    imported_modules = init_architecture(__name__, utils_modules, ui_modules)
  File "/Users/adamhill/Library/Application Support/Blender/2.90/scripts/addons/sverchok-master/core/__init__.py", line 110, in init_architecture
    import_all_modules(imported_modules, mods_bases)
  File "/Users/adamhill/Library/Application Support/Blender/2.90/scripts/addons/sverchok-master/core/__init__.py", line 95, in import_all_modules
    import_modules(mods, base, imported_modules)
  File "/Users/adamhill/Library/Application Support/Blender/2.90/scripts/addons/sverchok-master/core/__init__.py", line 65, in import_modules
    im = importlib.import_module('.{}'.format(m), base)
  File "/Applications/blender.app/Contents/Resources/2.90/python/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/adamhill/Library/Application Support/Blender/2.90/scripts/addons/sverchok-master/utils/sv_obj_baker.py", line 126, in <module>
    from sverchok.utils.solid import standard_mesher, mefisto_mesher
  File "/Users/adamhill/Library/Application Support/Blender/2.90/scripts/addons/sverchok-master/utils/solid.py", line 21, in <module>
    import MeshPart
ImportError: dlopen(/Users/adamhill/opt/anaconda3/envs/freecad37/lib/MeshPart.so, 2): Symbol not found: __ZN6netgen7testoutE
  Referenced from: /Users/adamhill/opt/anaconda3/envs/freecad37/lib/libNETGENPlugin.8.3.dylib
  Expected in: /Users/adamhill/opt/anaconda3/envs/freecad37/lib/libmesh.so
 in /Users/adamhill/opt/anaconda3/envs/freecad37/lib/libNETGENPlugin.8.3.dylib
```
I will make a PR maybe tomorrow or Tues for this issue and all the things I had to do to get geomdl to compile with Cython under Windows 10. (Missing Python development libs and headers, they were all in weird places and you need to tell the Python installer to install dev libs)

Thanks for this awesome tool. I hope to get all this working so I can get busy making some SVG's from 3D!

**Comment by enzyme69** [comment]
@adamhill Hi Adam, if you have a compiled FreeCAD that might work without crashing, I am also wanted to try it. If possible for you to temporary upload it?

**Comment by adamhill** [comment]
I didn't have to compile it, because we have **CONDA–FORGE**!!!

Follow the instructions [here on the Wiki](https://github.com/nortikin/sverchok/wiki/Dependencies#macos)

On the Mac I installed anaconda via ```Homebrew```. There are other methods out there if you don't use a package manager.

**Comment by foxxis** [comment]
@adamhill Sorry for commenting on a closed thread, but I have an identical problem to your initial problem: a crash on startup with freecad specified for sverchok.
The main difference is that I have followed the anaconda recipe and installed freecad for python 3.7.6 and also installed both the 0.19.x version and 0.18.4, and 0.18.2. I never get anything else than a crash.
Tried with all recent builds of sverchok,

Any tips would be most welcome!

`SciPy package is not available. Voronoi nodes and RBF-based nodes will not be available.
geomdl package is not available, some NURBS related nodes will not be available
SciKit-Image package is not available; SciKit-based implementation of Marching Cubes and Marching Squares will not be available
PyMCubes package is not available. PyMCubes-based implementation of Marching Cubes will not be available
Circlify package is not available. Circlify node will not be available
Writing: /var/folders/gh/t4hz1b516f181s3pz488wj3m0000gn/T/blender.crash.txt`
